manifests/init.pp

```diff
@@ -217,33 +203,21 @@
     }
   }
 
-  if $puppet {
+  # We can't pull the certs out to the top level, because of how it gets the default
+  # parameter values from the main ::certs class.  Kafo can't handle that case, so
+  # it remains here for now.
+  if defined(Class['puppet']) {
```

Comment:
This is dependent on the compile order so I'm not sure this is the best solution to detect if it was enabled.